app/actors/hyrax/actors/file_set_actor.rb

```diff
@@ -10,6 +10,40 @@ def initialize(file_set, user)
         @user = user
       end
 
+      # @!group Asynchronous Operations
+
+      # Spawns asynchronous IngestJob
+      # Called from FileSetsController, AttachFilesToWorkJob, ImportURLJob, IngestLocalFileJob
+      # @param [Hyrax::UploadedFile, File, ActionDigest::HTTP::UploadedFile, Tempfile] file the file uploaded by the user
```

Comment:
In practice, as currently used in the web app, we mostly get the fancier variety of params, so I'm OK with removing this from the YARD.  However, there is existing code referencing Tempfiles related to arkivo and `ImportUrlJob`.  If we want a technical fix here, it might involve: 
1. raise if given a Tempfile (even though on simplistic single-box inline systems, it would otherwise work)
1. automatically convert Tempfile to a `Hyrax::UploadedFile`
1. convert any remaining callers to passing non-Tempfile objects

I might be for some combination of 1 and 3 in a follow-up.

Note: there may be other kinds of "self destructing" file classes that will otherwise successfully pass through this method (and present the same issue jcoyne is talking about).